kernel/src/scheduler/round_robin.rs

```diff
@@ -120,9 +121,10 @@ impl<C: Chip> Scheduler<C> for RoundRobinSched<'_> {
             self.time_remaining.set(self.timeslice_length);
             self.timeslice_length
         };
-        assert!(timeslice != 0);
+        // Why should this panic?
+        let non_zero_timeslice = NonZeroU32::new(timeslice).unwrap();
 
-        SchedulingDecision::RunProcess((next, Some(timeslice)))
```

Comment:
Is this assert required in production code? It looks odd. I would rather add abstraction types such as `NonZeroU32` to show that timeslice may never be 0.

Reply:
I agree this code is really not what we would like. If `RoundRobinSched::time_remaining` was a `NonZeroU32` we could do proper checks and avoid the chance of a panic.